extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -0,0 +1,673 @@
+
+      Optional<SqlStatementResult> sqlStatementResult = getStatementStatus(
+          queryId,
+          authenticationResult.getIdentity(),
+          true
+      );
+
+      if (sqlStatementResult.isPresent()) {
+        return Response.ok().entity(sqlStatementResult.get()).build();
+      } else {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+    }
+    catch (DruidException e) {
+      return buildNonOkResponse(e);
+    }
+    catch (ForbiddenException e) {
+      log.debug("Got forbidden request for reason [%s]", e.getErrorMessage());
+      return buildNonOkResponse(
+          DruidException.forPersona(DruidException.Persona.USER)
+                        .ofCategory(DruidException.Category.FORBIDDEN)
+                        .build(Access.DEFAULT_ERROR_MESSAGE)
+      );
+    }
+    catch (Exception e) {
+      log.noStackTrace().warn(e, "Failed to handle query: %s", queryId);
+      return buildNonOkResponse(DruidException.forPersona(DruidException.Persona.DEVELOPER)
+                                              .ofCategory(DruidException.Category.UNCATEGORIZED)
+                                              .build(e, "Failed to handle query: [%s]", queryId));
+    }
+  }
+
+  @GET
+  @Path("/{id}/results")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response doGetResults(
+      @PathParam("id") final String queryId,
+      @QueryParam("offset") Long offset,
+      @QueryParam("numRows") Long numberOfRows,
+      @Context final HttpServletRequest req
+  )
+  {
+    try {
+      Access authResult = AuthorizationUtils.authorizeAllResourceActions(
+          req,
+          Collections.emptyList(),
+          authorizerMapper
+      );
+      if (!authResult.isAllowed()) {
+        throw new ForbiddenException(authResult.toString());
+      }
+      final AuthenticationResult authenticationResult = AuthorizationUtils.authenticationResultFromRequest(req);
+
+      if (offset != null && offset < 0) {
+        return buildNonOkResponse(
+            DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.INVALID_INPUT)
+                          .build(
+                              "offset cannot be negative. Please pass a positive number."
+                          )
+        );
+      }
+      if (numberOfRows != null && numberOfRows < 0) {
+        return buildNonOkResponse(
+            DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.INVALID_INPUT)
+                          .build(
+                              "numRows cannot be negative. Please pass a positive number."
+                          )
+        );
+      }
+
+      final long start = offset == null ? 0 : offset;
+      final long last = SqlStatementResourceHelper.getLastIndex(numberOfRows, start);
+
+
+      getStatementStatus(queryId, authenticationResult.getIdentity(), false);
```

Comment:
Is this call required? It seems to be checked after this already.

Reply:
Oh yes that needs to be removed. Thanks for the catch.